File: TransitAPI/TransitAPI/Models/BusLineType.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TransitAPI.Models
{
    public class BusLineType
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        [Required]
        [StringLength(256)]
        [Index(IsUnique = true)]        
        public string Name { get; set; }

        public List<BusLine> BusLines { get; set; }

    }
}

File: NodeJsApp/src/models/busLineType.js
import { Schema, model } from 'mongoose';

const busLineTypeSchema = new Schema({
    name: {
        type: String,
        required: true,
        unique: true
    },
    busLines: [
        {
            type: Schema.Types.ObjectId,
            ref: 'BusLine'
        }
    ]
});

export default model('BusLineType', busLineTypeSchema);